feat: Dynamic reviews, context batching, AWS Bedrock; GHES preserved; CLI+tests

m1rl0k opened the pull request:
Dynamic Reviews
Add custom prompt mode with customMode: on | off | auto
Auto mode routes complex code files to custom prompts
Capture documentation from first custom review batch and include in overview

Context Batching
Batch PR files by character size to respect config.maxReviewChars (default 725k)
Aggregate comments across batches

AWS Bedrock + SAP AI Core
Support Bedrock with IAM credentials (no LLM_API_KEY required when AWS creds present and model is bedrock/anthropic/meta/amazon family)
Keep SAP AI Core path (no API key required with proper SAP vars)

GitHub Enterprise Server (GHES) preserved
Inputs: github_api_url, github_server_url
Pass baseUrl to Octokit via initOctokit(config.githubToken, config.githubApiUrl)

CLI
Add --full (force full review) and --out (save output to file) flags

Comments improvements
Truncate code blocks in comments using config.maxCodeblockLines with marker: “… (truncated; more lines omitted) …”
Support multiline threads (start_line) in generateCommentThreads
Paginate listReviewComments to fetch all pages

Providers
AI SDK default temperature now 1 (when undefined)
Action + Config
action.yml: added inputs for new config (custom_mode, llm_provider, llm_model, review_scopes, allow_title_update, max_comments, max_codeblock_lines, max_review_chars)
branding: restored tags per request (ai, code-review, ai-assisted, automation, pull-requests, linting, ci, autofix, devtools)
Entry point remains dist/index.js

Build
Revert build script to two explicit esbuild outputs (dist/index.js, dist/cli.js)

Tests and Mocks
All tests green: 56 passed
Updated @octokit/action mock to expose Octokit.__lastOptions for throttle tests
Avoid custom prompt runner during tests to align with mocks

Misc
messages.ts: overview accepts optional documentation section

m1rl0k committed:
feat: Dynamic reviews, context batching, AWS Bedrock; GHES preserved; CLI+tests

Dynamic Reviews
Add custom prompt mode with customMode: on | off | auto
Auto mode routes complex code files to custom prompts
Capture documentation from first custom review batch and include in overview
Context Batching
Batch PR files by character size to respect config.maxReviewChars (default 725k)
Aggregate comments across batches
AWS Bedrock + SAP AI Core
Support Bedrock with IAM credentials (no LLM_API_KEY required when AWS creds present and model is bedrock/anthropic/meta/amazon family)
Keep SAP AI Core path (no API key required with proper SAP vars)
GitHub Enterprise Server (GHES) preserved
Inputs: github_api_url, github_server_url
Pass baseUrl to Octokit via initOctokit(config.githubToken, config.githubApiUrl)
CLI
Add --full (force full review) and --out (save output to file) flags
Comments improvements
Truncate code blocks in comments using config.maxCodeblockLines with marker: “… (truncated; more lines omitted) …”
Support multiline threads (start_line) in generateCommentThreads
Paginate listReviewComments to fetch all pages
Providers
AI SDK default temperature now 1 (when undefined)
Action + Config
action.yml: added inputs for new config (custom_mode, llm_provider, llm_model, review_scopes, allow_title_update, max_comments, max_codeblock_lines, max_review_chars)
branding: restored tags per request (ai, code-review, ai-assisted, automation, pull-requests, linting, ci, autofix, devtools)
Entry point remains dist/index.js
Build
Revert build script to two explicit esbuild outputs (dist/index.js, dist/cli.js)
Tests and Mocks
All tests green: 56 passed
Updated @octokit/action mock to expose Octokit.__lastOptions for throttle tests
Avoid custom prompt runner during tests to align with mocks
Removals
Removed push.ts and its tests per requirement
Misc
messages.ts: overview accepts optional documentation section

m1rl0k committed:
Update action.yml

bstanga commented on src/comments.ts:
```diff
@@ -97,5 +112,51 @@ export function isOwnComment(comment: string): boolean {
+    if (isFence(line)) {
+      if (inBlock) {
+        // closing fence
+        if (emittedTrunc) {
```
delete this if statement, it's no-op

bstanga commented:
Great PR! Do you have an example of a review with your changes applied?

bstanga reviewed (approved):
You need to rebase

m1rl0k committed:
Update comments.ts

m1rl0k committed:
Update prompts.custom.ts

m1rl0k committed:
build

m1rl0k committed:
test..

m1rl0k committed:
fix

m1rl0k committed:
test

m1rl0k committed:
anolther fix

type safety + schema validation...

m1rl0k committed:
Merge origin/main into feature-custom-prompts (keep our dist builds and @actions/github version)

m1rl0k committed:
bump github

m1rl0k commented:
Okay, again, maybe confirm... sorry for the issues

m1rl0k committed:
Update README.md

m1rl0k committed:
Update README.md

bstanga commented:
Sorry for delay, I've been on an extended leave. Will review this asap next week

bstanga commented on src/config.ts:
```diff
@@ -35,11 +41,52 @@ export class Config {
+      .filter((s) => !!s);
+
+    // Gate PR title updates (disabled by default per requirements)
+    const allowTitle = process.env.ALLOW_TITLE_UPDATE || getInput('allow_title_update') || 'false';
```
This is a breaking change for existing users, since right now this would be `true` by default.

Let's maybe keep it like that and introduce the option to disable it anything.

bstanga commented on src/config.ts:
```diff
@@ -120,6 +161,12 @@ export default process.env.NODE_ENV === "test"
+      customMode: "off",
+      reviewScopes: ["security", "performance", "best-practices"],
+      allowTitleUpdate: false,
+      maxComments: 40,
```
Are these two used anywhere? Just searching by them didn't give anything

m1rl0k committed:
Enable PR title update by default and remove max_comments

The action now allows PR title updates by default when @presubmit is present, updating the default in both action.yml and config.ts. The unused max_comments input and related logic have been removed for simplification.

m1rl0k committed:
Add support for custom LLM_BASE_URL for ai-sdk provider

Introduces the optional LLM_BASE_URL environment variable for OpenAI-compatible providers when using LLM_PROVIDER=ai-sdk. Updates documentation and configuration to clarify usage, enabling support for providers like OpenRouter, Anyscale, and Together AI. No changes for sap-ai-sdk provider.

m1rl0k commented:
@bstanga fixed both issues and merged your changes in from main

m1rl0k committed:
Refactor dotenv import and usage in CLI

Replaces legacy dotenv import and usage in dist/cli.js and related files with a new import style and updated variable names. This change improves consistency and maintainability by aligning with the latest build output and source conventions.

KingBain commented:
@m1rl0k  @bstanga  Could this be reopened and merged ?
